lr-cli/cortx_setup/commands/storage/config.py

```diff
@@ -259,205 +272,224 @@ def run(self, **kwargs):
+            if self.enclosure_id is not None and setup_type == "VM":
                 self.enclosure_id = "enc_" + self.machine_id
                 self.refresh_key_map()
                 self.store_in_file()
-                self.update_pillar_and_conf(
-                    'enclosure_id',
-                    self.enclosure_id
-                )
-            if not storage_type and not storage_type_in_conf:
-                #Set storage type to virtual for VM.
-                self.logger.debug(
-                    "Storage type is not provided, but since this is VM,"
-                    " setting the storage type to 'virtual'"
+                self.update_pillar_and_conf('enclosure_id', self.enclosure_id)
+
+        ### THE "mode" SHOULD ALWAYS BE THE FIRST CHECK, DO NOT CHANGE THIS SEQ ###
+        if self.mode is not None:
+            if ip is None or port is None:
+                # mandetory sub options for mode (ip and port) are missing.
+                self.logger.exception(
+                    f"Mandatory sub options for mode- ip & password are missing"
                 )
-                storage_type = "virtual"
-                self.update_pillar_and_conf('storage_type', storage_type)
-                storage_type = None
-            if name:
-                self.update_pillar_and_conf('name', name)
-            if storage_type:
-                if storage_type == "virtual":
-                    self.update_pillar_and_conf('storage_type', storage_type)
-                else:
-                    self.logger.error(
-                        "Storage type needs to be 'virtual' for VM")
-                    raise ValueError("Incorrect argument value provided")
-            if controller_type:
-                if controller_type == "virtual":
-                    self.update_pillar_and_conf(
-                        'controller_type',
-                        controller_type
+                raise RuntimeError('Please provide ip and port')
+            # Algorithm to update ip and port the enclosure id is needed.
+            # if enclosure_id and user and password
+            #    reset the enclosure id fetched from confstore
+            #    this is to force fetch the enclosure if with
+            #    current set of input parameters - user, passwd, ip, port.
+            # if not self.enclosure_id:
+            #     if hw:
+            #         if user and password:
+            #             #fetch enclosure id and store in confstore
+            #         else:
+            #             #error
+            # if self.enclosure_id:
+            #     # store ip and port in confstore
+            #     if user and password:
+            #         # store in confstore
+            if (
+                self.enclosure_id is not None
+                and user is not None
+                and password is not None
+            ):
+                # user has provided all the parameters that fetches the
```

Comment:
Strange condition here? Why would user still reset enclosure_id

Reply:
This is if user runs the same command again. In that case the enclosure id would be fetched from confstore and use, but to fetch the fresh enclosure id the enclosure id needs to be reset. This is what this condition is about.